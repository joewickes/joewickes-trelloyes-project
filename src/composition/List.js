import './List.css';
import Card from './Card';

function List(props) {
  const {header, cards} = props;

  // Map through the cards to return each one 

  return (
    <section className="List">
      <header className="List-header">
        <h2>{header}</h2>
      </header>
      <div className="List-cards">
        {cards.map((card) => <Card key={card.id} title={card.title} content={card.content} />)}
        <button type="button" className="List-add-button">
          + Add Random Card
        </button>
      </div>
    </section>
  );
}

export default List;