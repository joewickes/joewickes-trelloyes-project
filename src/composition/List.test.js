import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';

describe('List component', () => {
  it('renders List without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the Card as expected', () => {
    const tree = renderer
      .create(<List header="Title" cards={[
        {
          id: 1,
          title: 'a',
          content: 'b',
        },{
          id: 2,
          title: 'c',
          content: 'd',
        }]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});