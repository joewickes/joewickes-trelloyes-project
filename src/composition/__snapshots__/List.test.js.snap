// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the Card as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Title
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        a
      </h3>
      <p>
        b
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        c
      </h3>
      <p>
        d
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
